src/components/RouteUploadButtons.tsx

```diff
@@ -63,7 +61,17 @@ const RouteUploadButtons: VoidComponent<RouteUploadButtonsProps> = (props) => {
     } as Record<ButtonType, 'idle' | 'loading' | 'success' | 'error'>,
   })
 
-  const updateButtonStates = (types: ButtonType[], state: 'loading' | 'success' | 'error') => {
+  createEffect(() => {
+    if (props.route?.fullname === currentRoute()?.fullname) return
```

Comment:
You don't need a signal for this - effects will only run when their dependencies change

There's a helper function `on` you can use to explicitly define dependencies - https://docs.solidjs.com/reference/reactive-utilities/on-util

Keep in mind though that you are reading uploadStore.states so that probably adds a dependency - you could just create a list of the button types instead